sys-apps/razercfg/razercfg-0.42.ebuild

```diff
@@ -0,0 +1,96 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit cmake python-single-r1 systemd tmpfiles udev
+
+DESCRIPTION="Utility for advanced configuration of Razer mice"
+HOMEPAGE="https://bues.ch/cms/hacking/razercfg.html"
+SRC_URI="https://bues.ch/${PN}/${P}.tar.xz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="qt5 +udev"
```

Comment:
There is no configure option for `qt5` flag, but instead you're doing removals based on it? That doesn't sound okay. If they're required, they're required.

Reply:
That does look strange, but this is how razercfg is.

This is a mouse configuration utility with an optional python qt5 gui. The gui is indeed not required and is cleanly disabled by removing related files. Having a strict qt5 dependency for a mouse configuration utility is... wrong. Even arch linux folks do not do this: https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/razercfg

It would be better if there was a configuration option not to install the gui, but there isn't one.

Reply:
What if we didn't remove anything, but added `dev-python/PyQt5` as `optfeature` to this ebuild? I'd say that's a better solution here.

Reply:
It is possible to add it as an `optfeature`, but I don't think it is very consistent:
* without pyqt5, we would install a desktop app and applet which fail to work
* if you want gui, you need pyqt5 as a part of your world set; when there is qt6, you would have to update it; looking at the world file you would not remember why you have pyqt5 there

In arch linux they do it as an `optfeature`, but that is because they do not have a better way like use flags.

P.S.: Also, on the bugzilla I offered to proxy maintain this package, at least while I use a razer mouse myself.

Reply:
> 
>     * without pyqt5, we would install a desktop app and applet which fail to work
> 

It'd probably give a descriptive error message, and make the user remember post-merge message, or at least to peek inside the ebuild.

>     * if you want gui, you need pyqt5 as a part of your world set; when there is qt6, you would have to update it; looking at the world file you would not remember why you have pyqt5 there
> 
> 

This can be managed, 
https://wiki.gentoo.org/wiki//etc/portage/sets (check optfeature example)

> In arch linux they do it as an `optfeature`, but that is because they do not have a better way like use flags.
> 

We don't use USE flags for this.
https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301
https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001

Also for future just FYI it should probably be `gui` USE flag, 
https://projects.gentoo.org/qa/policy-guide/use-flags.html

Reply:
I agree that `gui` would be better than `qt5`.

> > ```
> > * without pyqt5, we would install a desktop app and applet which fail to work
> > ```
> 
> It'd probably give a descriptive error message, and make the user remember post-merge message, or at least to peek inside the ebuild.

It does give a message `...No module named 'PyQt5.QtCore'`, but it does not appear if the user clicks the app icon.

> > ```
> > * if you want gui, you need pyqt5 as a part of your world set; when there is qt6, you would have to update it; looking at the world file you would not remember why you have pyqt5 there
> > ```
> 
> This can be managed,
> https://wiki.gentoo.org/wiki//etc/portage/sets (check optfeature example)

This surely can be managed, but it is more complicated than a use flag, and there is no sufficiently good reason for complexity.

> https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0301
From the first sentence of description, this does not appy: "Ebuilds must not introduce USE flags to control installing files that are small in size, require no additional dependencies and not cause any negative consequences to the program behavior by being installed." The GUI files do require an additional dependency, and do cause negative consequences by installing programs and desktop icon which do not work.

> https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001
It is not clear if this applies either: with the GUI files installed, pyqt5 is not an optional runtime dependency, but rather a normal dependency. It is something without which the core works fine but other parts fail with an error. In addition to this, rebuild of razercfg is quick and cheap, so the rationale does not apply.

I think that the right solution would be to ask the upstream to add a configuration option for GUI. In the absence of that, a use flag is a simple and very standard solution. An optfeature is not too bad either, but without pyqt5 it installs broken programs, including a desktop icon, which is not friendly.

Reply:
Okay, let's make it `gui` USE flag. Looking at recent upstream issues the main maintainer isn't writing anything for this anymore, and if you want something implemented you need to write it on your own. Wouldn't matter though since nothing is done in `src_configure` or `_compile` regarding the outcome of these files.